axelor-sale/src/main/java/com/axelor/apps/sale/rest/SaleOrderRestController.java

```diff
@@ -63,4 +67,19 @@ public Response createSaleOrder(SaleOrderPostRequest requestBody)
 
     return ResponseConstructor.buildCreateResponse(saleOrder, new SaleOrderResponse(saleOrder));
   }
+
+  @Operation(
+      summary = "Create a sale oder",
+      tags = {"Sale order"})
+  @Path("/")
+  @PUT
+  @HttpExceptionHandler
+  public Response changeSaleOrderStatus(SaleOrderPutRequest requestBody) throws AxelorException {
+    RequestValidator.validateBody(requestBody);
+    new SecurityCheck().writeAccess(SaleOrder.class, requestBody.getSaleOrderId()).check();
+
+    Beans.get(SaleOrderGeneratorService.class)
+        .changeSaleOrderStatus(requestBody.fetchSaleOrder(), requestBody.getStatusId());
+    return ResponseConstructor.build(Response.Status.OK, I18n.get(ITranslation.STATUS_CHANGE));
```

Comment:
Please wait for ticket 83791 to be merged and you will be able to call SaleOrderFinalizeService#finalizeQuotation and SaleOrderConfirmService#confirmProcess directly from the controller.

You can check TimesheetRestController#updateTimesheetStatus for an example.